app/gui/src/dashboard/layouts/Settings/FormEntry.tsx

```diff
@@ -0,0 +1,67 @@
+/** @file Rendering for an {@link SettingsFormEntryData}. */
+import { ButtonGroup, Form } from '#/components/AriaComponents'
+import { useSyncRef } from '#/hooks/syncRefHooks'
+import { useText } from '#/providers/TextProvider'
+import { useEffect, useMemo, useState } from 'react'
+import SettingsInput from './Input'
+import type { SettingsContext, SettingsFormEntryData } from './data'
+
+// =========================
+// === SettingsFormEntry ===
+// =========================
+
+/** Props for a {@link SettingsFormEntry}. */
+export interface SettingsFormEntryProps<T extends Record<keyof T, string>> {
+  readonly context: SettingsContext
+  readonly data: SettingsFormEntryData<T>
+}
+
+/** Rendering for an {@link SettingsFormEntryData}. */
+export function SettingsFormEntry<T extends Record<keyof T, string>>(
+  props: SettingsFormEntryProps<T>,
+) {
+  const { context, data } = props
+  const { schema: schemaRaw, getValue, inputs, onSubmit, getVisible } = data
+  const { getText } = useText()
+  const [dirty, setDirty] = useState(false)
+  const visible = getVisible?.(context) ?? true
+  const value = getValue(context)
+  const schema = useMemo(
+    () => (typeof schemaRaw === 'function' ? schemaRaw(context) : schemaRaw),
+    [context, schemaRaw],
+  )
+
+  const form = Form.useForm({
+    // @ts-expect-error This is SAFE, as the type `T` is statically known.
+    schema,
+    defaultValues: value,
+    onSubmit: async (newValue) => {
+      // @ts-expect-error This is SAFE, as the type `T` is statically known.
+      await onSubmit(context, newValue)
+      setDirty(true)
```

Comment:
`getValue` is currently synchronous so we will get the old value i think.

should we maybe instead just `form.reset(newValue)`?
also should the form be inlined again in this case?

Reply:
> also should the form be inlined again in this case?

I'm completely unopinionated here.

> should we maybe instead just form.reset(newValue)?

Yep, like that, the issue is the `useState` + `useEffect` instead of calling the `form.reset()` directly